src/components/05_pages/Permissions/Permissions.js

```diff
@@ -1,17 +1,98 @@
-import React from 'react';
-import { css } from 'emotion';
+import React, { Fragment, Component } from 'react';
+import Loading from '../../Helpers/Loading';
 
-const styles = {
-  title: css`
-    text-decoration: underline;
-  `,
+const Permissions = class Permissions extends Component {
+  state = {
+    loaded: false,
+  };
+  componentDidMount() {
+    Promise.all([
+      fetch(
+        `${process.env.REACT_APP_DRUPAL_BASE_URL}/permissions?_format=json`,
+      ).then(res => res.json()),
+      fetch(
+        `${process.env.REACT_APP_DRUPAL_BASE_URL}/jsonapi/user_role/user_role`,
+      ).then(res => res.json()),
+    ])
+      .then(([permissions, { data: roles }]) =>
+        this.setState({ permissions, roles, loaded: true }),
+      )
+      .catch(err => this.setState({ loaded: false, err }));
+  }
+  groupPermissions = permissions =>
+    Object.entries(
+      Object.keys(permissions)
+        .map(key => permissions[key])
+        .reduce((acc, cur) => {
+          acc[cur.provider] = acc[cur.provider] || [];
+          acc[cur.provider].push(cur);
+          return acc;
+        }, {}),
+    );
+  render() {
+    return (
+      <Fragment>
+        {!this.state.loaded ? (
```

Comment:
Here is a generic question. Is there a reason why you do the condition inside the fragment and not outside of it?

Reply:
Since we're just returning the `<table />` here we probably don't even need the fragment at all.